### DISCUSSION TITLE
如何在modal.confirm的onCanel中判断是点击取消按钮还是点击右上角的X？

### DISCUSSION THREAD

**Discussion by Dmaziyo** [ant-design/ant-design#51851] [Q&A] [answered]
const showConfirm = () => {
    confirm({
        title: 'Do you want to delete these items?',
        icon: React.createElement(ExclamationCircleFilled, null),
        content: 'Some descriptions',
        closable: true,
        onOk() {
            console.log('OK');
        },
        onCancel(e) {
            console.log('Cancel');
        },
    });
};

**Comment by Dmaziyo** [comment]
![image](https://github.com/user-attachments/assets/acfcb7af-c871-4834-9790-7bfc2eb53009)
场景如上图所示

**Comment by zombieJ** [comment, marked as answer]
footer 可以自定义，自己做个取消按钮自行监听事件。那剩下的就是x了

**Reply by Dmaziyo** [reply]
我刚刚试了下，发现还可以根据返回的函数名称来判断
![image](https://github.com/user-attachments/assets/c4a9b572-8aff-4518-8b85-3213d212b483)

**Reply by afc163** [reply]
e.name 会被压缩，最好用 e.target 判断。
